crates/amaru-ledger/src/state.rs

```diff
@@ -364,26 +369,53 @@ impl<S: Store, HS: HistoricalStores> State<S, HS> {
     }
 }
 
-#[allow(clippy::panic)]
-fn recover_stake_distribution(
+// NOTE: Initialize stake distribution held in-memory. The one before last is needed by the
+// consensus layer to validate the leader schedule, while the one before that will be
+// consumed for the rewards calculation.
+//
+// We always hold on two stake distributions:
+//
+// - The one from an epoch `e - 1` which is used for the ongoing leader schedule at epoch `e + 1`
+// - The one from an epoch `e - 2` which is used for the rewards calculations at epoch `e + 1`
+//
+// Note that the most recent snapshot we have is necessarily `e`, since `e + 1` designates
+// the ongoing epoch, not yet finished (and so, not available as snapshot).
+pub fn stake_distributions(
```

Comment:
```suggestion
pub fn initial_stake_distributions(
```

This work is really only meant to happen once, on start-up.